### PR TITLE
fix build on clang 7

### PR DESCRIPTION
**Pull request by sethalves**
- fix build on clang 7

### CODE REVIEW

**Comment by sethalves** [comment]
I don't think this one requires QA

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13750/HighFidelity-Beta-PR14173-f947583.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13750/HighFidelity-Beta-PR14173-f947583.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13750/HighFidelity-Beta-Interface-PR14173-f947583.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13750/HighFidelity-Beta-PR14173-f947583.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13750/HighFidelity-Beta-Interface-PR14173-f947583.exe

File: assignment-client/src/audio/AudioMixerClientData.cpp [hunk 1]
```diff
@@ -497,7 +497,12 @@ void AudioMixerClientData::processStreamPacket(ReceivedMessage& message, Concurr
 
     if (newStream) {
         // whenever a stream is added, push it to the concurrent vector of streams added this frame
+#if (defined(__clang__) && defined(Q_OS_LINUX))
+        addedStreams.push_back(AddedStream(getNodeID(), getNodeLocalID(), matchingStream->getStreamIdentifier(), matchingStream.get()));
+#else
         addedStreams.emplace_back(getNodeID(), getNodeLocalID(), matchingStream->getStreamIdentifier(), matchingStream.get());
+#endif
```
Comment (Atlante45) [inline assignment-client/src/audio/AudioMixerClientData.cpp]: We should use `push_back` everywhere.
This adds complexity for very little gain.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13754/HighFidelity-Beta-PR14173-de8ae91.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13754/HighFidelity-Beta-PR14173-de8ae91.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13754/HighFidelity-Beta-Interface-PR14173-de8ae91.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13754/HighFidelity-Beta-PR14173-de8ae91.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13754/HighFidelity-Beta-Interface-PR14173-de8ae91.exe

**Comment by qliejahner** [comment]
Currently testing

**Comment by qliejahner** [comment]
PASS @conklin94122 

Build installed and launched without issue

**Pull request merged**